examples/config.hocon.sample

```diff
@@ -193,6 +193,14 @@ collector {
     # request can be cached. -1 seconds disables the cache. Chromium max is 10m, Firefox is 24h.
     accessControlMaxAge = 5 seconds
     accessControlMaxAge = ${?COLLECTOR_CORS_ACCESS_CONTROL_MAX_AGE}
+
+    # Whether or not the value in `restrictDomainsTo` is respected. Optional, defaults to true.
```

Comment:
If `restrictDomainsTo` is omitted the default behavior is to allow all domains. Right now, with an empty list the behavior would be to disallow all domains, which I don't think would ever be useful. I see what you mean when you say we could reduce the possibility of misconfiguration by saying that an omitted list and an empty list both allow all domains. I didn't do this because I was worried it could also break user expectations (e.g. that removing a domain from the list could never make the system more permissive).

However, we can go through with your idea if you think the trade-off makes sense. If we do this I agree that `restrictedDomins` would be a better name.

I want to explain the motivation for the `doRestrictDomains` parameter. I saw that many of the other sub-configurations have a `enable` or `enabled` boolean parameter that just allows the user to turn those variables on or off. I wanted the new configuration to have a similar option, but naming it `enabled` would have been misleading since it doesn't have any effect on the behavior of the other parameter in `cors` that was already there (`accessControlMaxAge`).

Being able to disable fields with a bool parameter has the advantage of not requiring the user to delete the existing contents (for example, perhaps the user is repeatedly modifying the config, and wants to disable the behavior of the field temporarily for testing purposes, but intends to restore it later). I suppose it could also be accomplished by commenting out the relevant lines, but then the commented parts would not be checked for syntax or type errors. This is also why it defaults to `true` — having the functionality of `restrictDomainsTo` disabled by default would probably be more confusing.

So it's true that `doRestrictDomains` is not strictly necessary, even with the current design, it was just there because I thought it would be more consistent with the `enabled` fields elsewhere in the file. Again, I'll defer to you if you don't think this is valuable.

Reply:
Yes, I believe a single option that lets users pass in a list of domains would be best.

We have three cases:
- We want to allow one of more domains (but in the event only one of them will be returned in the ACAO header because of the limitations of how it works -- `ACAO(origin)`)
- We want to allow all domains (`*`)
- We want to disallow all domains (no ACAO header).

This is tricky to express unambiguously in the config but I think the best would be:

```
# Allow one or more domains
allowedDomains = [ "myfirstdomain.com", "myseconddomain.com", ...]

# Disallow all domains
allowedDomains = [ ]

# Allow all domains by deleting the option
# allowedDomains
```

This is still a bit finicky, with having to delete the option to allow `*` but I think it's preferable to something like  `restrictedDomains`. With that, we'd still need to delete the option to allow `*` but we'll also have `restrictedDomains  = [ ]` meaning that no domains will be allowed, while it looks like no domains will be restricted.

I understand your motivation for `doRestrict` and appreciate the effort to make the config consistent. I think in this case the benefits of simplicity outweigh the downsides.

Reply:
I completely agree with @dilyand on this one